spec/rspec/core/bisect/example_minimizer_spec.rb

```diff
@@ -19,6 +19,81 @@ module RSpec::Core
       expect(minimizer.repro_command_for_currently_needed_ids).to eq("rspec ex_2 ex_4 ex_5")
     end
 
+    it 'reduces a failure where none of the passing examples are implicated' do
+      no_dependents_runner = FakeBisectRunner.new(
+        %w[ ex_1 ex_2 ],
+        %w[ ex_2 ],
+        {}
+      )
+      minimizer = Bisect::ExampleMinimizer.new(no_dependents_runner, RSpec::Core::NullReporter)
+      minimizer.find_minimal_repro
+      expect(minimizer.repro_command_for_currently_needed_ids).to eq("rspec ex_2")
+    end
+
+    it 'reduces a failure when more than 50% of examples are implicated' do
+      fake_runner.always_failures = []
+      fake_runner.dependent_failures = { "ex_8" => %w[ ex_1 ex_2 ex_3 ex_4 ex_5 ex_6 ] }
+      minimizer = Bisect::ExampleMinimizer.new(fake_runner, RSpec::Core::NullReporter)
+      minimizer.find_minimal_repro
+      expect(minimizer.repro_command_for_currently_needed_ids).to eq(
+        "rspec ex_1 ex_2 ex_3 ex_4 ex_5 ex_6 ex_8"
+      )
+    end
+
+    it 'reduces a failure with multiple dependencies' do
+      fake_runner.always_failures = []
+      fake_runner.dependent_failures = { "ex_8" => %w[ ex_1 ex_3 ex_5 ex_7 ] }
+      minimizer = Bisect::ExampleMinimizer.new(fake_runner, RSpec::Core::NullReporter)
+      minimizer.find_minimal_repro
+      expect(minimizer.repro_command_for_currently_needed_ids).to eq(
+        "rspec ex_1 ex_3 ex_5 ex_7 ex_8"
+      )
+    end
+
+    context 'with an unminimisable failure' do
+      class RunCountingReporter < RSpec::Core::NullReporter
+        attr_accessor :round_count
+        attr_accessor :example_count
+        def initialize
+          @round_count = 0
+        end
+
+        def publish(event, *args)
+          send(event, *args) if respond_to? event
+        end
+
+        def bisect_individual_run_start(_notification)
+          self.round_count += 1
+        end
+      end
+
+      let(:counting_reporter) { RunCountingReporter.new }
+      let(:fake_runner) do
+        FakeBisectRunner.new(
+          %w[ ex_1 ex_2 ex_3 ex_4 ex_5 ex_6 ex_7 ex_8 ex_9 ],
+          [],
+          "ex_9" => %w[ ex_1 ex_2 ex_3 ex_4 ex_5 ex_6 ex_7 ex_8 ]
+        )
+      end
+      let(:counting_minimizer) do
+        Bisect::ExampleMinimizer.new(fake_runner, counting_reporter)
+      end
+
+      it 'returns the full command if the failure can not be reduced' do
+        counting_minimizer.find_minimal_repro
+
+        expect(counting_minimizer.repro_command_for_currently_needed_ids).to eq(
+          "rspec ex_1 ex_2 ex_3 ex_4 ex_5 ex_6 ex_7 ex_8 ex_9"
+        )
+      end
+
+      it 'detects an unminimisable failure in the minimum number of runs' do
+        counting_minimizer.find_minimal_repro
+
+        expect(counting_reporter.round_count).to eq(15)
```

Comment:
It's unclear from this example why 15 is the minimum number of runs.  It kinda feels like a magic number right now. Can you add a comment (or update the docstring, if the explanation is short enough) explaining why 15 is the minimum?